packages/docgen/src/engine.js

```diff
@@ -19,25 +18,21 @@ const getExportTokens = ( ast ) =>
 			'ExportNamedDeclaration',
 			'ExportDefaultDeclaration',
 			'ExportAllDeclaration',
-		].some( ( declaration ) => declaration === node.type )
+		].includes( node.type )
 	);
 
 const engine = ( path, code, getIRFromPath = () => {} ) => {
-	const result = {};
-	result.ast = getAST( code );
-	result.tokens = getExportTokens( result.ast );
-	result.ir = flatten(
-		result.tokens.map( ( token ) =>
-			getIntermediateRepresentation(
-				path,
-				token,
-				result.ast,
-				getIRFromPath
-			)
-		)
+	if ( ! path || ! code ) {
+		return null;
```

Comment:
This changes the output of `engine` for consumers. We should either maintain the output or adapt https://github.com/WordPress/gutenberg/blob/master/packages/docgen/src/index.js#L54 to work with nulls.

Reply:
OK, I reverted the return value of `processFile` to always be an object. In case of error, it will return `{ ast: null, tokens: null, ir: [] }`. That's what the consumers and also what the tests expect.